packages/auth/amplify_auth_cognito_dart/lib/src/auth_plugin_impl.dart

```diff
@@ -183,6 +184,9 @@ class AmplifyAuthCognitoDart extends AuthPluginInterface<
   DeviceMetadataRepository get _deviceRepo =>
       _stateMachine.getOrCreate(DeviceMetadataRepository.token);
 
+  /// Analytics Metadata Provider
+  AnalyticsMetadataTypeBuilder? get _analyticsMetadata => _stateMachine.get();
```

Comment:
Agreed but the requests it takes all require the builder object instead of the actual instance, so when feeding into the requests I'd have to say something like _analyticsMetadata.toBuilder()

Reply:
That's the best way to do it. A builder is only a temporary object for updating a value before concretizing it. It shouldn't be long-lasting since its mutable.

Reply:
Okay I used this form since it was shorter: 
`          ..analyticsMetadata = _analyticsMetadata?.toBuilder();
`